packages/owl-bot/src/owl-bot.ts

```diff
@@ -76,15 +78,29 @@ export = (privateKey: string | undefined, app: Probot, db?: Db) => {
       if (!installation) {
         throw Error(`no installation token found for ${head.repo.full_name}`);
       }
+      const hasRunLabel = !!context.payload.pull_request.labels.filter(
+        l => l.name === OWLBOT_RUN_LABEL
+      ).length;
+
       // If the pull request is from a fork, the label "owlbot:run" must be
       // added by a maintainer to trigger the post processor.
-      // TODO(bcoe): add support for "owlbot:run" label.
-      if (head.repo.full_name !== base.repo.full_name) {
+      if (head.repo.full_name !== base.repo.full_name && !hasRunLabel) {
         logger.info(
           `head ${head.repo.full_name} does not match base ${base.repo.full_name} skipping`
         );
         return;
       }
+      // If the pull request is *not* from a fork, ignore label being added
+      // (we will be running OwlBot post processor for the push itself);
+      if (
+        head.repo.full_name === base.repo.full_name &&
+        context.payload.action === 'labeled'
```

Comment:
Suggestion: if there's separate validation based on the event type, use separate probot `app.on()` handlers and move the common behavior to a helper function.

Reply:
I've split into two `on`s, I think it simplifies logic.